hibernate-core/src/main/java/org/hibernate/resource/transaction/backend/jdbc/internal/JdbcResourceLocalTransactionCoordinatorImpl.java

```diff
@@ -84,8 +84,7 @@ public void explicitJoin() {
 
 	@Override
 	public boolean isJoined() {
-		return physicalTransactionDelegate != null && physicalTransactionDelegate.getStatus() == TransactionStatus.ACTIVE;
-
+		return physicalTransactionDelegate != null && getTransactionDriverControl().isActive( true );
```

Comment:
This code is for JDBC based transactions.  Your test tests a JTA use-case.

Even though JPA does not allow it, I thought we allowed `#joinTransaction` to be called for JDBC ("resource local") transactions.